my_compassion/controllers/my2_children.py

```diff
@@ -21,56 +22,133 @@ def my2_render_children_page(self, **kwargs):
         partner = request.env.user.partner_id
 
         breadcrumbs = [
-            {'name': 'Children', 'url': '/my2/children/', 'active': True},
+            {"name": "Children", "url": "/my2/children/", "active": True},
         ]
 
-        return request.render(
-            'my_compassion.my2_children_page',
+        sponsorships_data = [] 
+
+        # Iterate through each sponsorship of the partner
+        for sponsorship in partner.sponsorship_ids:
+            has_unread_letter = bool(
+                request.env["correspondence"].search(
+                    [
+                        ("child_id", "=", sponsorship.child_id.id),
+                        ("email_read", "=", False),
+                        ("direction", "=", "Beneficiary To Supporter"),
+                    ],
+                    limit=1,
+                )
+            )
+
+            # Calculate days until next birthday
+            today = datetime.today().date()
+            birthday_this_year = sponsorship.child_id.birthdate.replace(year=today.year)
+            delta = (birthday_this_year - today).days
+            # If the birthday has already occurred this year, calculate for next year
+            if delta < 0:
+                birthday_next_year = birthday_this_year.replace(year=today.year + 1)
+                delta = (birthday_next_year - today).days
+
+            # Calculate days since last letter
+            last_letter = request.env["correspondence"].search(
+                [
+                    ("child_id", "=", sponsorship.child_id.id),
+                    ("direction", "=", "Supporter To Beneficiary"),
+                ],
+                order="create_date DESC",
+                limit=1,
+            )
+
+            if last_letter:
+                last_letter_date = last_letter.create_date.date()
+                delta_last_letter = (today - last_letter_date).days
+
+        sponsorships_data.append(
```

Comment:
This should be done for each loop iteration (i.e. for all sponsored children.) Current indentation makes that only the last iteration is taken into account and displayed.

```suggestion
             sponsorships_data.append(
```